engine/src/flutter/lib/web_ui/lib/src/engine/safe_browser_api.dart

```diff
@@ -11,100 +11,41 @@
-}
-
-const Set<String> _safeJsProperties = <String>{'decoding', '__flutter_state'};
-
-/// Sets the value of property [name] on a JavaScript [object].
-///
-/// This is equivalent to writing `object.name = value` in plain JavaScript.
-T setJsProperty<T>(Object object, String name, T value) {
-  assert(
-    _safeJsProperties.contains(name),
-    'Attempted to set property "$name" on a JavaScript object. This property '
-    'has not been checked for safety. Possible solutions to this problem:\n'
-    ' - Do not set this property.\n'
-    ' - Use a `js_util` API that does the same thing.\n'
-    ' - Ensure that the property is safe then add it to _safeJsProperties set.',
-  );
-  return js_util.setProperty<T>(object, name, value);
-}
-
-/// Converts a JavaScript `Promise` into Dart [Future].
-Future<T> promiseToFuture<T>(Object jsPromise) {
-  return js_util.promiseToFuture<T>(jsPromise);
-}
 
 /// Parses a string [source] into a double.
 ///
 /// Uses the JavaScript `parseFloat` function instead of Dart's [double.parse]
 /// because the latter can't parse strings like "20px".
 ///
-/// Returns null if it fails to parse.
-num? parseFloat(String source) {
+/// Returns `null` if it fails to parse.
+double? parseFloat(String source) {
   // Using JavaScript's `parseFloat` here because it can parse values
   // like "20px", while Dart's `double.tryParse` fails.
-  final num? result = js_util.callMethod(domWindow, 'parseFloat', <Object>[source]);
+  final double result = parseFloatImpl(source.toJS).toDartDouble;
 
-  if (result == null || result.isNaN) {
+  if (result.isNaN) {
     return null;
   }
   return result;
 }
 
-/// Used to decide if the browser tab still has the focus.
-///
-/// This information is useful for deciding on the blur behavior.
-/// See [DefaultTextEditingStrategy].
-///
-/// This getter calls the `hasFocus` method of the `Document` interface.
-/// See for more details:
-/// https://developer.mozilla.org/en-US/docs/Web/API/Document/hasFocus
-bool get windowHasFocus => js_util.callMethod<bool>(domDocument, 'hasFocus', <dynamic>[]);
-
 /// Parses the font size of [element] and returns the value without a unit.
 num? parseFontSize(DomElement element) {
   num? fontSize;
 
-  if (hasJsProperty(element, 'computedStyleMap')) {
+  if (element.hasProperty('computedStyleMap'.toJS).toDart) {
     // Use the newer `computedStyleMap` API available on some browsers.
-    final Object? computedStyleMap = js_util.callMethod<Object?>(
-      element,
-      'computedStyleMap',
-      const <Object?>[],
-    );
-    if (computedStyleMap is Object) {
-      final Object? fontSizeObject = js_util.callMethod<Object?>(computedStyleMap, 'get', <Object?>[
-        'font-size',
-      ]);
-      if (fontSizeObject is Object) {
-        fontSize = js_util.getProperty<num>(fontSizeObject, 'value');
+    final computedStyleMap = element.callMethod<JSObject?>('computedStyleMap'.toJS);
+    if (computedStyleMap != null) {
```

Comment:
Bonus points if we add types for `computedStyleMap` and `fontSizeObject`, so we get a cool one-line nesting and avoid `dart:js_interop_unsafe` altogether e.g. `element.computedStyleMap()?.get('font-size')?.value`.

Reply:
`value` is...weird. I did the best I could.